Rubberduck.Parsing/Symbols/VariableDeclaration.cs

```diff
@@ -51,6 +50,15 @@ public sealed class VariableDeclaration : Declaration, IInterfaceExposable
             }
         }
 
+        /// <summary>
+        /// True if a variable is declared with the <c>Static</c> keyword, or declared in a procedure scope that uses this keyword.
+        /// </summary>
+        /// <remarks>
+        /// In VBA/VB6, a Static variable keeps its value between procedure calls.
+        /// </remarks>
+        public bool IsStatic => ParentScopeDeclaration is ModuleBodyElementDeclaration parent && parent.IsStatic
+                                || Context.Parent.Parent is VBAParser.VariableStmtContext context && context.STATIC() != null;
```

Comment:
Why add this property? Whether a local variable is static is stored in its `Accessability`.

Reply:
@MDoerner what's the `Accessibility` of a `Public Static Sub` procedure? Not sure `Static` really belongs in the `Accessibility` enum. Or it can work, but only if we use the member-level modifier to make locals have `Static` accessibility (i.e. no member shall have `Static` accessibility). Thoughts?

Reply:
I'm keeping the `IsStatic` property on `ModuleBodyElementDeclaration`, as it's not really any different than the `Block` property on the same class: we're getting a specific member from a specific `ParserRuleContext`, and this allows a new `DeclarationSymbolsListener.GetLocalVariableAccessibility` method to easily query `_currentScopeDeclaration.IsStatic` and thus know whether a local variable is implicitly `Static`.